crates/router/src/core/payment_methods/utils.rs

```diff
@@ -0,0 +1,857 @@
+                            .into_iter()
+                            .map(dir::DirValue::BillingCountry)
+                            .collect();
+
+                        let config_country_agg_node = builder
+                            .make_in_aggregator(dir_countries, None, None::<()>)
+                            .map_err(KgraphError::GraphConstructionError)?;
+
+                        let node = builder
+                            .make_all_aggregator(
+                                &[
+                                    (
+                                        pm_object_country_value_node,
+                                        cgraph::Relation::Negative,
+                                        cgraph::Strength::Strong,
+                                    ),
+                                    (
+                                        config_country_agg_node,
+                                        cgraph::Relation::Positive,
+                                        cgraph::Strength::Strong,
+                                    ),
+                                ],
+                                None,
+                                None::<()>,
+                                None,
+                            )
+                            .map_err(KgraphError::GraphConstructionError)?;
+                        return Ok(Some(node));
+                    }
+                }
+            }
+        }
+        admin::AcceptedCountries::AllAccepted => return Ok(None),
+    }
+    Ok(None)
+}
+
+fn compile_accepted_currency_for_mca(
+    builder: &mut cgraph::ConstraintGraphBuilder<'_, dir::DirValue>,
+    payment_method_type: &enums::PaymentMethodType,
+    pm_obj_currency: admin::AcceptedCurrencies,
+    config: &settings::ConnectorFilters,
+    connector: String,
+) -> Result<Option<cgraph::NodeId>, KgraphError> {
+    match pm_obj_currency {
+        admin::AcceptedCurrencies::EnableOnly(currency) => {
+            if let Some(config) = config
+                .0
+                .get(connector.as_str())
+                .or_else(|| config.0.get("default"))
+            {
+                if let Some(value) =
+                    config
+                        .0
+                        .get(&settings::PaymentMethodFilterKey::PaymentMethodType(
+                            *payment_method_type,
+                        ))
+                {
+                    // Currency from the MCA
+                    let pm_object_currency_value_node = builder
+                        .make_in_aggregator(
+                            currency
+                                .into_iter()
+                                .map(dir::DirValue::PaymentCurrency)
+                                .collect(),
+                            None,
+                            None::<()>,
+                        )
+                        .map_err(KgraphError::GraphConstructionError)?;
```

Comment:
Same here. Probably should be constructed regardless of whether the config exists or not.